### PR TITLE
fix: Add power check in updateValset() of Peggy.sol

### PR DESCRIPTION
**Pull request by Hsiang-xxs**

### CODE REVIEW

**Comment by zmitton** [comment]
looks right to me

File: solidity/contracts/Peggy.sol [hunk 1]
```diff
@@ -276,6 +276,19 @@ uint256 public state_powerThreshold;
       "Supplied current validators and powers do not match checkpoint."
     );
 
+    uint256 state_powerThreshold_ = state_powerThreshold;
```
Comment (facundomedica) [inline solidity/contracts/Peggy.sol:279]: Q: why is this necessary?

Comment (Hsiang-xxs) [reply solidity/contracts/Peggy.sol:279]: To save gas fee

File: solidity/contracts/Peggy.sol [hunk 2]
```diff
@@ -276,6 +276,19 @@ uint256 public state_powerThreshold;
...
+			if (cumulativePower > state_powerThreshold_) {
+				break;
+			}
+		}
+		require(
+			cumulativePower > state_powerThreshold_,
+			"New validator set signatures do not have enough power." 
+		);
+
     // Check that enough current validators have signed off on the new validator set
```
Comment (facundomedica) [inline solidity/contracts/Peggy.sol:292]: nit: fix indentation and some spacing

```suggestion
    uint256 cumulativePower;

    for (uint256 i = 0; i < _newValset.powers.length; i++) {
        cumulativePower = cumulativePower + _newValset.powers[i];
        if (cumulativePower > state_powerThreshold_) {
            break;
        }
    }

    require(
        cumulativePower > state_powerThreshold_,
        "New validator set signatures do not have enough power." 
    );

    // Check that enough current validators have signed off on the new validator set
```

Comment (Hsiang-xxs) [reply solidity/contracts/Peggy.sol:292]: Yep, correct 👍

File: solidity/contracts/Peggy.sol (hunk 1 above)
Comment (facundomedica) [reply solidity/contracts/Peggy.sol:279]: Awesome then! 😄

**Review by facundomedica — APPROVED** [review]

**Review by alexanderbez — APPROVED** [review]
